strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -198,11 +180,11 @@ async def handle_connection_init(self, message: ConnectionInitMessage) -> None:
             return
 
         self.connection_init_received = True
-        await self.send_message(ConnectionAckMessage())
+        await self.websocket.send_json(ConnectionAckMessage({"type": "connection_ack"}))
```

Comment:
As much as I like the improvement stemming from the dicts above, I really dislike this redundancy of types. What about adding a new `create` `classmethod` on the `TypedDict` class that handles filling this value? Not very pythonic, but at least you don't need to pass the message types again

Reply:
Similar story to the previous comment: At runtime TypedDicts are just regular dicts. When subclassing `TypedDict` we can't add any custom methods (or constants) to it.
While this new style looks more verbose, it's also more explicit since it's no longer hiding away any message fields. I honestly like it, especially since it makes pyright and mypy super happy.

Here's something we could do with the hightlighted code:
`self.websocket.send_json` takes any dict as argument, that's why we wrap our message dict in `ConnectionAckMessage` so that the type checker ensures it has a valid shape.
Instead we could add and call a (new) `send_message(self, message: Message)` method, which expects a valid `Message` as argument. In the example above, we would no longer need to wrap our typed dict in `ConnectionAckMessage`, because the type checker would ensure that the message's shape is valid based on it's `type` field.

So instead of:

`await self.websocket.send_json(COnnectionAckMessage({"type": "connection_ack"})`

we would write:

`await self.send_message({"type": "connection_ack"})`

and type checker would still provide autocompletion and enforce that messages have a valid shape.

Reply:
(Just added two commits doing exactly that)

Reply:
My issue was mainly with the redundancy and verbosity of `COnnectionAckMessage({"type": "connection_ack"})`, but when reviewing I forgot to touch the possibility of just sending a normal dict :D Happy with this solution